vyxal/lexer.py

```diff
@@ -100,32 +99,33 @@ def __repr__(self) -> str:
         Returns
         -------
         str
-            [name, value]
+            Token(name, value)
         """
 
-        return str([self.name.value, self.value])
+        return f"Token({repr(self.name.value)}, {repr(self.value)})"
 
-    def __eq__(self, rhs: Token) -> bool:
+    def __eq__(self, rhs) -> bool:
         """
-        Returns whether both tokens have the same attributes, because
+        Return whether both tokens have the same attributes because
         memory addresses won't be the same.
 
         Parameters
         ----------
 
-        rhs : Token
-            The token to compare.
+        rhs : object
+            An object to compare this token with.
 
         Returns
         -------
 
-        True iff the two token names and values are the same.
+        True iff `rhs` is a token and has the same name and value.
         """
-
+        if not isinstance(rhs, Token):
+            return False
```

Comment:
I believe standard Python practice would be to return `NotImplemented` for other types, so the other object can decide whether it's comparable